common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/BrowserAuthorizationFragment.java

```diff
@@ -94,6 +94,12 @@ public void onSaveInstanceState(@NonNull Bundle outState) {
 
     @Override
     void extractState(final Bundle state){
+        if (state == null) {
```

Comment:
Should we do this in the base class?

Reply:
It is already done there but is not sufficient as that method returns back here

Reply:
super is called first, which does this logic and returns. Then we are back here, try to use the state param, it is NULL and we get NPE

Reply:
ahh okay.

In the base class, can we move the logic to onCreate instead? (before extractState is called).

Thanks for catching the bug btw.

Reply:
Why do you want to finish in the onCreate? The state param in onCreate could be NULL and is currently handled by the `mInstanceState` instance variable. See this: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/744fb75a6d443aef8ee11c88a22886128cd4a9f9/common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationFragment.java#L84-L101

Reply:
We could make extractState() takes NonNull parameter only, and then put the check inside onCreate(). We're already doing half the check there (if savedInstanceState==null) anyway.

I just prefer having the check in one place - it's easier to maintain.

Reply:
Addressed